PluginCode/src/main/java/smr/cs/ualberta/libcomp/action/ReplacementAction.java

```diff
@@ -209,51 +222,65 @@ public void detectDependancy(@NotNull final Editor editor, @NotNull final PsiFil
-                            depObj.setEnableddomain(false);
-                        }
-                        else {
-                            depObj.setEnableddomain(true);
-                        }
+                selectedTerm = valuesInQuotes[0];
 
-                        DependListObjects.add(depObj);
+                DatabaseAccess dataAccessObject = new DatabaseAccess();
+                ArrayList<String> choicesArray = dataAccessObject.selectJsonAllLibraries(selectedTerm);
 
+                if (choicesArray.size() > 0) {
+                    DependencyStatement depObj = new DependencyStatement();
+                    depObj.setImportLocation(i);
+                    depObj.setImportLib(Integer.parseInt(choicesArray.get(0)));
+                    depObj.setImportDomain(Integer.parseInt(choicesArray.get(1)));
+                    depObj.setDomainName(choicesArray.get(2));
+                    depObj.setFromlocation(startOffset);
+                    depObj.setTolocation(endOffset);
 
-                        if (depObj.getEnableddomain()) {
-                            editorModel.addLineHighlighter(i,
-                                    DebuggerColors.BREAKPOINT_HIGHLIGHTER_LAYER + 1, softerAttributes);
+                    if (dataAccessObject.isEnabled(Integer.parseInt(choicesArray.get(1)), projectName)) {
+                        depObj.setEnableddomain(false);
+                    }
+                    else {
+                        depObj.setEnableddomain(true);
+                    }
 
-                        }
+                    listObjects.add(depObj);
 
+                    if (depObj.getEnableddomain()) {
+                        editorModel.addLineHighlighter(i,
+                                DebuggerColors.BREAKPOINT_HIGHLIGHTER_LAYER + 1, softerAttributes);
 
                     }
                 }
+            }
             ++i;
-            boolean isContainsEnd = lineText.contains("}");
+            boolean isContainsEnd = lineText.contains(endIdentifier);
             if (isContainsEnd) {
                 dependenciesExists = false;
             }
         }
     }
 
+    public void detectDependancy(@NotNull final Editor editor, @NotNull final PsiFile psiFile, @NotNull final  String projectName ) throws IOException {
+
+        final Document document = editor.getDocument();
+        int lineNum = 0;
+        int loc = detectDependenciesPSI(psiFile.getNode()); // Parse PSI to detect the PSI dependencies node
+        if (loc != -1) // dependencies exists
+        {
+            lineNum = document.getLineNumber(loc); // line number of the dependencies PSI node
+        }
+
+        try {
+            detectStatement(editor, projectName, loc, lineNum, "\'", "\'","}", DependListObjects);
+        }
+        catch (IOException ioException) {
+            ioException.printStackTrace();
+        }
+    }
+
 
     public int detectMavenPSI(final Document document)
```

Comment:
I'm confused between all the detect functions... can you add a method comment above each method to briefly describe its purpose

Reply:
I made comments for all detect functions